build_automation/frontend/src/js/addTag.js

```diff
@@ -39,15 +49,47 @@ class TagCreation extends React.Component {
         var targetUrl = this.props.listUrl;
         const payload = { name: this.state.name, description: this.state.description };
         const currentInstance = this;
-        {
+        if (this.state.id > 0) {
+            // Update an exising Tag.
+            targetUrl = get_url(this.props.detailUrl, { id: this.state.id });
+            axios.patch(targetUrl, payload, {
+                responseType: 'json'
+            }).then(function (response) {
+                currentInstance.saveCallback(response.data, TAG_SAVE_TYPE.UPDATE);
+            }).catch(function (error) {
+                console.error("Error in updating the tag.", error);
+                console.error(error.response.data);
+                let errorMsg = 'Error in creating the tag.';
+            if (error.response.status === HTTP_STATUS.BAD_REQUEST) {
+                errorMsg = (<React.Fragment><b>ERROR:</b> This metadata already exists. Please rename it before trying to update.</React.Fragment>);
```

Comment:
IMO "Please rename it before trying to update" is not needed, since the metadata already exists says it.